flink-runtime/src/main/java/org/apache/flink/runtime/metrics/MetricRegistry.java

```diff
@@ -144,6 +152,30 @@ public MetricRegistry(Configuration config) {
 		}
 	}
 
+	/**
+	 * Initializes the MetricQueryService.
+	 * 
+	 * @param actorSystem ActorSystem to create the MetricQueryService on
+     */
+	public void startQueryService(ActorSystem actorSystem) {
+		try {
+			queryService = MetricQueryService.startMetricQueryService(actorSystem);
+		} catch (Exception e) {
+			LOG.warn("Could not start MetricDumpActor. No metrics will be submitted to the WebInterface.", e);
+		}
+	}
+
+	/**
+	 * Returns an ActorRef to the MetricQueryService
+	 * 
+	 * @return ActorRef to the MetricQueryService
+     */
+	public Option<ActorRef> getQueryService() {
+		return queryService == null
```

Comment:
this is equivalent to `Option.apply(queryService)`. If queryService == null, then the apply call will return a `None`.